substrate/executor/src/wasm_executor.rs

```diff
@@ -352,6 +353,15 @@ impl_function_executor!(this: FunctionExecutor<'e, E>,
 		this.memory.set(out, &result).map_err(|_| UserError("Invalid attempt to set result in ext_blake2_256"))?;
 		Ok(())
 	},
+	ext_keccak256(data: *const u8, len: u32, out: *mut u8) => {
+		let result = if len == 0 {
+			keccak256(&[0u8; 0])
+		} else {
+			keccak256(&this.memory.get(data, len as usize).map_err(|_| UserError("Invalid attempt to get data in ext_keccak256"))?)
```

Comment:
unfortunate that we have to reallocate the data from memory. but it's a limitation of WASM AFAIK now